packages/webapp/src/components/Animals/DetailCards/Origin.tsx

```diff
@@ -116,14 +136,28 @@ const Origin = ({ t, currency, originOptions, namePrefix = '' }: OriginProps) =>
       </>
     );
   }, [origin, Object.entries(errors)]);
-
   return (
     <div className={styles.sectionWrapper}>
       {/* @ts-ignore */}
       <Input
         type="date"
         label={t('ANIMAL.ATTRIBUTE.DATE_OF_BIRTH')}
-        hookFormRegister={register(`${namePrefix}${DetailsFields.DATE_OF_BIRTH}`)}
+        hookFormRegister={register(`${namePrefix}${DetailsFields.DATE_OF_BIRTH}`, {
+          validate: (value) => {
+            if (value === '' && isBirthDateValid) return true;
+            return moment(value, 'YYYY-MM-DD', true).isValid()
+              ? isNotInFuture(value)
+              : t('common:INVALID_DATE');
+          },
+        })}
+        onCleared={() => {
+          setIsBirthDateValid(true);
+          clearErrors(`${namePrefix}${DetailsFields.DATE_OF_BIRTH}`);
+        }}
+        onKeyUp={(e: any) => {
```

Comment:
Not a blocker for merge but if we could annotate these functions properly like so

```suggestion
        onKeyUp={(e: React.ChangeEvent<HTMLInputElement>): void => {
```

it'd be great!